Formula/imgproxy.rb

```diff
@@ -0,0 +1,28 @@
+class Imgproxy < Formula
+  desc "Fast and secure server for resizing and converting remote images"
+  homepage "https://imgproxy.net"
+  url "https://github.com/imgproxy/imgproxy/archive/v2.7.0.tar.gz"
+  sha256 "9a97a4b184f67718f97a4b3d16cd98b61a825ba418e6e04139ed3a3a9d787fad"
+  head "https://github.com/imgproxy/imgproxy.git"
+
+  depends_on "go" => :build
+  depends_on "pkg-config" => :build
+  depends_on "vips"
+
+  def install
+    ENV.prepend_path "PKG_CONFIG_PATH", Formula["libffi"].opt_lib/"pkgconfig"
+    ENV["CGO_LDFLAGS_ALLOW"]="-s|-w"
+    ENV["CGO_CFLAGS_ALLOW"]="-Xpreprocessor"
+
+    system "go", "build", "-o", "#{bin}/#{name}"
+  end
+
+  test do
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
This is an HTTP server that resize images on the fly.
So the ideal test would probably be running it as a daemon, point it to a local directory with at least one image, and make an HTTP request for this image.
Do you know any other formula that does HTTP requests as part of testing I can use as base?

Reply:
httpd is a server